static/js/stream_edit.js

```diff
@@ -594,6 +596,31 @@ export function initialize() {
         settings_ui.do_settings_change(channel.patch, url, data, $status_element);
     }
 
+    $("#manage_streams_container").on("click", ".copy_email_button", (e) => {
+        const stream_id = get_stream_id(e.target);
+        const stream = sub_store.get(stream_id);
+        const address = stream.email_address;
+
+        const copy_email_address = render_copy_email_address_modal({
+            email_address: address,
+        });
+
+        dialog_widget.launch({
+            html_heading: $t_html({defaultMessage: "Generate stream email address"}),
+            html_body: copy_email_address,
+            id: "copy_email_address_modal",
+            html_submit_button: "Copy address",
+            on_click: () => {},
+            close_on_submit: true,
+        });
+
+        new ClipboardJS(".dialog_submit_button", {
```

Comment:
Not sure if this is a problem, but the CSS selector looks too general / not specific enough.

Reply:
This might be fine -- one can only have one `dialog_widget` visible at a time.

@sahil839 @ganpa3 FYI -- do we have a cleaner way to do this?

Reply:
@juliaBichler01 You can use the `id` of the modal to be more specific here
```
#copy_email_address_modal .dialog_submit_button
```